.github/scripts/check_base_image_tags/check_base_image_tags.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+"""
+Check that base_image references to pipelines-components images use :main tag.
+
+This ensures components reference the latest images rather than specific SHAs or versions.
+The CI will override these with PR-specific tags during validation.
+"""
+
+import argparse
+import re
+import sys
+from pathlib import Path
+
+
+def find_base_image_references(file_path: Path, image_prefix: str) -> list[tuple[int, str]]:
+    """
+    Find lines containing base_image references to the given image prefix.
+
+    Args:
+        file_path: Path to the Python file to scan.
+        image_prefix: The image prefix to look for (e.g., ghcr.io/kubeflow/pipelines-components).
+
+    Returns:
+        List of tuples (line_number, line_content) for matching lines.
+
+    Raises:
+        OSError: If the file cannot be read.
+        UnicodeDecodeError: If the file contains invalid encoding.
+    """
+    matches = []
+    content = file_path.read_text()
+    for line_num, line in enumerate(content.splitlines(), start=1):
+        if "base_image" in line and image_prefix in line:
```

Comment:
couldn't this get some false positives, like comments or even unrelated variables?

for example
```
check_if_base_image_is_python = do_something()
```
as this would satisfy the requirements of having both "base_image" and a image name if it's common enough (like a basic `python` image off dockerhub)

It would be uncommon but i think we need to use ast parsing here to guarantee we're looking at base_image parameter values of a decorator and not unrelated code that could fail.